### PR TITLE
switch GCR -> Artifact-Registry

### PR DESCRIPTION
**Pull request by ccwienk**
GCR has been deprecated [0] in favour of Artifact-Registry.

Thus, change push-targets for OCI-Images:

- europe-docker.pkg.dev/gardener-project/snapshots for snapshots
- europe-docker.pkg.dev/gardener-project/releases for releases
- europe-docker.pkg.dev/gardener-project/public for combined view of snapshots + releases

[0]
https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

**Release note**:

```breaking operator
Change OCI Image Registry from GCR (`eu.gcr.io/gardener-project`) to Artifact-Registry (`europe-docker.pkg.dev/gardener-project/releases`). Users should update their references.

```

### CODE REVIEW

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -12,7 +10,6 @@ gardener-extension-runtime-gvisor:
         - linux/arm64
         dockerimages:
           gardener-extension-runtime-gvisor:
-            registry: 'gcr-readwrite'
             image: 'eu.gcr.io/gardener-project/gardener/extensions/runtime-gvisor'
```
Comment (MrBatschner) [inline .ci/pipeline_definitions:13]: Shouldn't this be updated to point to `europe-docker.pkg.dev/gardener-project/releases`?

File: .ci/pipeline_definitions [hunk 2]
```diff
@@ -26,7 +23,6 @@ gardener-extension-runtime-gvisor:
               - type: 'githubUser'
                 username: 'danielfoehrKn'
           gardener-extension-runtime-gvisor-installation:
-            registry: 'gcr-readwrite'
             image: 'eu.gcr.io/gardener-project/gardener/extensions/runtime-gvisor-installation'
```
Comment (MrBatschner) [inline .ci/pipeline_definitions:26]: see above

File: .ci/pipeline_definitions (hunk 1 above)
Comment (ccwienk) [reply .ci/pipeline_definitions:13]: oh dear. It seems I missed the second commit on that pull-request :-/

Comment (ccwienk) [reply .ci/pipeline_definitions]: pushed the second commit as well (I already had it in my local repository) :-)

**Review by MrBatschner — APPROVED** [review]
/lgtm

**Pull request merged**